src/Operators/finitedifference.jl

```diff
@@ -1459,6 +1459,139 @@ end
     return Geometry.Contravariant3Vector(vᶠ * a)
 end
 
+"""
+    U = FCTBorisBook(;boundaries)
+    U.(v, x)
+
+Correct the flux using the flux-corrected transport formulation by Boris and Book
+Input arguments:
+- a face-valued vector field `v`
+- a center-valued field `x`
+```math
+U(v,x)[i] = \\begin{cases}
+  v[i] \\left(-2 x[i-\\tfrac{3}{2}] + 10 x[i-\\tfrac{1}{2}] + 4 x[i+\\tfrac{1}{2}] \\right) / 12  \\textrm{, if } v[i] > 0 \\\\
+  v[i] \\left(4 x[i-\\tfrac{1}{2}] + 10 x[i+\\tfrac{1}{2}] -2 x[i+\\tfrac{3}{2}]  \\right) / 12  \\textrm{, if } v[i] < 0
+  \\end{cases}
+```
+This stencil is based on [BorisBook1973](@cite).
+
+Supported boundary conditions are:
+- [`ThirdOrderOneSided(x₀)`](@ref): uses the third-order downwind reconstruction to compute `x` on the left boundary,
+and the third-order upwind reconstruction to compute `x` on the right boundary.
+
+!!! note
+    Similar to the [`Upwind3rdOrderBiasedProductC2F`](@ref) operator, these boundary conditions do not define the value at the actual boundary faces, and so this operator cannot be materialized directly: it needs to be composed with another operator that does not make use of this value, e.g. a [`DivergenceF2C`](@ref) operator, with a [`SetValue`](@ref) boundary.
+"""
+struct FCTBorisBook{BCS} <: AdvectionOperator
+    bcs::BCS
+end
+FCTBorisBook(; kwargs...) = FCTBorisBook(NamedTuple(kwargs))
+
+return_eltype(::FCTBorisBook, V, A) =
+    Geometry.Contravariant3Vector{eltype(eltype(V))}
+
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceFiniteDifferenceSpace,
+    arg_space::Spaces.CenterFiniteDifferenceSpace,
+) = velocity_space
+return_space(
+    ::FCTBorisBook,
+    velocity_space::Spaces.FaceExtrudedFiniteDifferenceSpace,
+    arg_space::Spaces.CenterExtrudedFiniteDifferenceSpace,
+) = velocity_space
+
+@inline function fct_boris_book(v, a⁻, a⁺, a⁺⁺, a⁺⁺⁺, step)
+    if v != zero(eltype(a⁻))
+        sign(v) ⊠ (RecursiveApply.rmap(
```

Comment:
Does your v represent A in section 5.4.1, does your a represent phi^td ? Is step = dx/dt? You can change it to if v < zero(eltype(a⁻))

Reply:
Yes, A in section 5.4.1 has the role of `v` here (defined at faces, like the velocity) and `a` is phi^td in Sec. 5.4.1. Step is only the spatial step size, so only `dz`

Reply:
> You can change it to if v < zero(eltype(a⁻))

This should be the same as what is there, I've opened a [comment](https://github.com/CliMA/ClimaCore.jl/pull/883#discussion_r944626258) regarding the floating point comparison.